components/gitpod-cli/cmd/ports-list.go

```diff
@@ -59,7 +64,7 @@ var listPortsCmd = &cobra.Command{
 					status = "detecting..."
 					statusColor = tablewriter.FgYellowColor
 				}
-			} else if port.Served {
+			} else if port.Served && port.Exposed != nil {
```

Comment:
Should not it be `served` if exposed is nil?

Reply:
I've refactored the code